proof/invariant-abstract/ARM/ArchCNodeInv_AI.thy

```diff
@@ -222,20 +222,7 @@ lemma vs_cap_ref_update_cap_data[simp, CNodeInv_AI_assms]:
                 arch_update_cap_data_def
          split: cap.split)
 
-
-lemma in_preempt[simp, intro, CNodeInv_AI_assms]:
-  "(Inr rv, s') \<in> fst (preemption_point s) \<Longrightarrow>
-  (\<exists>f es. s' = s \<lparr> machine_state := machine_state s \<lparr> irq_state := f (irq_state (machine_state s)) \<rparr>, exst := es\<rparr>)"
-  apply (clarsimp simp: preemption_point_def in_monad do_machine_op_def
-                        return_def returnOk_def throwError_def o_def
-                        select_f_def select_def getActiveIRQ_def)
-  done
-
-
-lemma invs_irq_state_independent[intro!, simp, CNodeInv_AI_assms]:
-  "invs (s\<lparr>machine_state := machine_state s\<lparr>irq_state := f (irq_state (machine_state s))\<rparr>\<rparr>)
-   = invs s"
-  by auto
+lemmas [CNodeInv_AI_assms] = invs_irq_state_independent
```

Comment:
Yes. You can see about 100 instances in l4v.

Reply:
Thanks. Does it have a different effect from `declare`?

Reply:
To add a bit more here, this style is generally used to add attributes to multiple existing lemmas at the same time. However, it can also be used to create an unnamed lemma and give it attributes.

We generally don't like this second use case since it is impossible to remove those attributes later on (for instance, if we needed to remove the lemma from the simp set). However, there might be some edge cases for specific uses such as the arch_split assumptions.

Reply:
right, yes, I think I actually wrote this line and I did so without thinking too much, probably simply because you see this pattern sometimes particularly for named_theorem bundles for locale instantiation, just like here. And, as you say, this is one of the uses that are fairly ok to do this.